eos-enable-zram

```diff
@@ -3,22 +3,98 @@
 # Licensed under the GPLv2
 
 if [ $# -lt 1 -o "$1" -lt 0 -o "$1" -gt 1 ]; then
-	echo "Error: 0 or 1 must be specified" >&2
-	exit 1
+    echo "Error: 0 or 1 must be specified" >&2
+    exit 1
 fi
 
+if [ "$1" -eq 0 ]; then
+    swapoff /dev/zram0
+    echo 1 > /sys/block/zram0/reset
+    exit 0
+fi
+
+# List and disable any active swap areas
+active_swaps=($(swapon --show=NAME --raw --noheadings))
+echo "Active swap areas before $(basename $0): ${active_swaps[*]}"
+swapoff -a
+
+# Enable zram
 ram_size_kb=$(awk '/MemTotal/{print $2}' /proc/meminfo)
+if [ ! -e /sys/block/zram0 ]; then
+    modprobe zram
+fi
+echo $(($ram_size_kb * 2))K > /sys/block/zram0/disksize
+mkswap /dev/zram0
+swapon /dev/zram0
 
-if [ "$1" -ne 0 ]; then
-    if [ ! -e /sys/block/zram0 ]; then
-        modprobe zram
+if [ -z "${active_swaps}" ]; then
+    exit 0
+fi
+
+# Find root partition, root partition number, root disk, and table type
+orig_root_part=$(systemctl show -p What sysroot.mount)
+orig_root_part=${orig_root_part#What=}
+root_part=$(readlink -f "${orig_root_part}")
+case ${root_part} in
+    /dev/sd??)
+        root_disk=${root_part%?}
+        ;;
+    /dev/*p[0-9])
+        root_disk=${root_part%p?}
+        ;;
+esac
+
+case ${orig_root_part} in
+    /dev/mapper/endless-image?)
+        root_disk=${orig_root_part%?}
+        ;;
+esac
+
+if [ -z "${root_disk}" ]; then
+    exit 0
+fi
+
+for swap_area in "${active_swaps[@]}" ; do
```

Comment:
On a dual-boot system with swap, are the active swaps listed as `/dev/dm-X` or `/dev/mapper/endless-image-deviceX`? If the latter you'll need to duplicate the special handling of that case here.

Reply:
I'll need to check that, I don't have a dual-boot system available atm.